src/tools/message_filter.rs

```diff
@@ -0,0 +1,135 @@
+//! Message filtering types and utilities.
```

Comment:
I'm just assumming this was copied verbatim? i.e. moved

Reply:
I removed some functions and the logging which became unused as a result of the synth node refactor, functionality is much the same.